src/detectors/FHCAL/FHCAL.cc

```diff
@@ -3,60 +3,208 @@
+             "HcalEndcapPInsertHits"},                     // edm4hep::SimCalorimeterHitCollection
+            {"HcalEndcapPInsertTruthClusters",             // edm4eic::Cluster
+             "HcalEndcapPInsertTruthClusterAssociations"}, // edm4eic::MCRecoClusterParticleAssociation
+            {
+              .energyWeight = "log",
+              .moduleDimZName = "",
+              .sampFrac = 1.0,
+              .logWeightBase = 3.6,
+              .depthCorrection = 0.0,
+              .enableEtaBounds = true
+            },
+            app   // TODO: Remove me once fixed
+          )
+        );
+
+        app->Add(
+          new JChainMultifactoryGeneratorT<Cluster_factory_HcalEndcapPInsertClusters>(
+             "HcalEndcapPInsertClusters",
+            {"HcalEndcapPInsertIslandProtoClusters",  // edm4eic::ProtoClusterCollection
+             "HcalEndcapPInsertHits"},                // edm4hep::SimCalorimeterHitCollection
+            {"HcalEndcapPInsertClusters",             // edm4eic::Cluster
+             "HcalEndcapPInsertClusterAssociations"}, // edm4eic::MCRecoClusterParticleAssociation
+            {
+              .energyWeight = "log",
+              .moduleDimZName = "",
+              .sampFrac = 1.0,
+              .logWeightBase = 6.2,
+              .depthCorrection = 0.0,
+              .enableEtaBounds = false,
+            },
+            app   // TODO: Remove me once fixed
+          )
+        );
 
         app->Add(new JFactoryGeneratorT<RawCalorimeterHit_factory_LFHCALRawHits>());
         app->Add(new JFactoryGeneratorT<CalorimeterHit_factory_LFHCALRecHits>());
         app->Add(new JFactoryGeneratorT<ProtoCluster_factory_LFHCALTruthProtoClusters>());
         app->Add(new JFactoryGeneratorT<ProtoCluster_factory_LFHCALIslandProtoClusters>());
-        app->Add(new JFactoryGeneratorT<Cluster_factory_LFHCALClusters>());
-        app->Add(new JFactoryGeneratorT<Cluster_factory_LFHCALTruthClusters>());
 
+        app->Add(
+          new JChainMultifactoryGeneratorT<Cluster_factory_LFHCALTruthClusters>(
+             "LFHCALTruthClusters",
+            {"LFHCALTruthProtoClusters",        // edm4eic::ProtoClusterCollection
+             "LFHCALHits"},                     // edm4hep::SimCalorimeterHitCollection
+            {"LFHCALTruthClusters",             // edm4eic::Cluster
+             "LFHCALTruthClusterAssociations"}, // edm4eic::MCRecoClusterParticleAssociation
+            {
+              .energyWeight = "log",
+              .moduleDimZName = "",
+              .sampFrac = 1.0,
+              .logWeightBase = 4.5,
+              .depthCorrection = 0.0,
+              .enableEtaBounds = false
+            },
+            app   // TODO: Remove me once fixed
+          )
+        );
+
+        app->Add(
+          new JChainMultifactoryGeneratorT<Cluster_factory_LFHCALClusters>(
+             "LFHCALClusters",
+            {"LFHCALIslandProtoClusters",  // edm4eic::ProtoClusterCollection
```

Comment:
```suggestion
            {"LFHCALProtoClusters",  // edm4eic::ProtoClusterCollection
```
as in https://github.com/eic/EICrecon/pull/666/files#diff-65cecdd85beb3ba502721f7ec59a332712e214e8c85a8b4804af074edfabe929L55

Reply:
I mean https://github.com/eic/EICrecon/pull/666/files#diff-803292d5047f9f39733fc1e4c25e513ffd878b83d7bd3419daaf5c50f2c3896eL56

Reply:
This doesn't seem right because (on `main`):
https://github.com/eic/EICrecon/blob/main/src/detectors/FHCAL/ProtoCluster_factory_LFHCALIslandProtoClusters.h

Reply:
The actual inconsistency seems to be in the name `EcalBarrelSciGlassProtoClusters`